feat: Update live events filters to match mockups

alexkim205 opened the pull request:
## Problem

Fixes https://github.com/PostHog/posthog/issues/8798



https://user-images.githubusercontent.com/13460330/157360060-63632fda-1630-4589-9764-f014162d4f6f.mov






## Changes

- [x] Shows delete button when user can clear event filter.
- [x] Update live events filters

👉 *Stay up-to-date with our [coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

alexkim205 committed:
allow event filter to be clearable

clarkus reviewed (dismissed):
Thanks for getting this change in. If there is capacity for a quick iteration, I would propose we use a pattern that feels less destructive. Typically the trash can icons are reserved for actions that would take some effort to recover from... for example clearing a graph series from a complex insight, deleting anything, etc. In this case, the recovery is simple, so an x icon is used within the bounds of the input to indicate the scope of the action and make it feel less-destructive and more encapsulated. 

<img width="277" alt="Screen Shot 2022-03-08 at 2 27 53 PM" src="https://user-images.githubusercontent.com/254612/157336482-0cfe7c57-e4bc-4180-96fa-45a3b80a5bcf.png">

https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=6989%3A37786

alexkim205 committed:
Merge remote-tracking branch 'origin' into fix/events-filter

rcmarron reviewed (approved):
LGTM!

alexkim205 committed:
new live events look

alexkim205 committed:
fix visibility

alexkim205 commented:
Addressed feedback! Figured I might as well update the entire Live Events filters to match the mockups. Updated the PR description with the latest version.

mariusandra reviewed (dismissed):
Looks good, though:

1. The wrapping could use a few tweaks on certain resolutions, and with elements of certain lengths:
![2022-03-09 09 47 49](https://user-images.githubusercontent.com/53387/157405717-3db14eb5-3a03-4fc8-9878-7efa3bf77a41.gif)

2. Those grey pills look out of place now. I assume that's out of scope for now?

Twixes commented on frontend/src/lib/components/PropertyFilters/components/FilterRow.tsx:
```diff
@@ -126,6 +128,23 @@ export const FilterRow = React.memo(function FilterRow({
+                                                    color="var(--primary)"
+                                                    style={{ marginRight: '-0.25rem' }}
+                                                    fontSize={14}
+                                                />
```
It's cleaner to use only CSS props for styling instead of some HTML (`color` and `fontSize` here) and some CSS.
But regarding font size and margin, it seems in the design it's the default – let's not override them for individual buttons unless completely necessary, as then various inconsistencies creep in.

Twixes commented on frontend/src/lib/components/TaxonomicPopup/TaxonomicPopup.tsx:
```diff
@@ -81,3 +84,103 @@ export function TaxonomicPopup({
+                                <LemonButton
+                                    className="side-buttons-row-button"
+                                    type="tertiary"
+                                    icon={<IconClose fontSize={16} />}
```
The `LemonButton` prop `compact` is the canonical way of getting slightly smaller buttons for places like this.

alexkim205 replied:
Compact is pretty cool but I don't think this is what I'm trying to achieve here. I want to make the icon itself smaller, but compact only shrinks the height of the hoverable area.

Twixes commented on frontend/src/lib/components/icons.tsx:
```diff
@@ -1239,3 +1239,25 @@ export function IconCorporate(props: React.SVGProps<SVGSVGElement>): JSX.Element
     )
 }
+
+export function IconTuning(props: React.SVGProps<SVGSVGElement>): JSX.Element {
+    return (
+        <svg width="18" height="18" viewBox="0 0 18 18" fill="none" xmlns="http://www.w3.org/2000/svg" {...props}>
+            <path
+                d="M0 14V16H6V14H0ZM0 2V4H10V2H0ZM10 18V16H18V14H10V12H8V18H10ZM4 6V8H0V10H4V12H6V6H4ZM18 10V8H8V10H18ZM12 6H14V4H18V2H14V0H12V6Z"
+                fill="currentColor"
+            />
+        </svg>
+    )
+}
+
+export function IconExport(props: React.SVGProps<SVGSVGElement>): JSX.Element {
+    return (
+        <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg" {...props}>
```
Icons should be standardized on `width="1em" height="1em"` (ideally also `viewBox="0 0 24 24"`).
Also, I wonder where they come from? Most of our icons come from Material Design now, so they've got comments like `/** Material Design Refresh icon. */`. This export one looks a lot like Material "File Download", is that correct?

alexkim205 replied:
oo nice catch, these are definitely material icons. I'd just copied them over from Figma but didn't know their source. Updating these

Twixes commented on frontend/src/lib/components/LemonButton/LemonButton.scss:
```diff
@@ -1,6 +1,5 @@
 .LemonButton {
     color: var(--primary);
-    font-weight: 600;
```
This should still be the case!

alexkim205 replied:
Hm, just curious what is the reason we enforce this on all buttons? It seems like something that should be off by default since a lot of the buttons in the Figma mockups have normal font weights.

Twixes commented on frontend/src/lib/components/LemonButton/LemonButton.scss:
```diff
@@ -94,4 +93,27 @@
         right: 1.25rem;
         transform: translate(50%, -50%);
     }
+    .side-buttons-row {
+        display: flex;
+        flex-direction: row;
+        align-items: center;
+
+        .side-buttons-row-button {
+            position: relative;
+
+            &.side-buttons-row-button-no-hover {
+                &:not(:disabled):hover {
+                    background: unset;
+                }
+            }
+        }
+
+        .side-buttons-row-button-divider {
+            width: 1px;
+            height: 20px;
+            margin-left: 0.25rem;
+            margin-right: 0.25rem;
+            background-color: var(--border);
+        }
+    }
```
These styles don't seem to belong in this file, as they're only used by `TaxonomicPopup.tsx`. They should probably instead be specifically scoped for `TaxonomicPopup` and sit in `TaxonomicPopup.scss`.

Twixes reviewed (dismissed):
This is a good direction, but some fixes are needed.
Most importantly, this results in unintended changes to buttons in various places! I commented in a few files about this.
Also, I agree with Marius that things look weird with filters added. Do we have a design for an alternative for the pills to be implemented in a follow-up? More immediately, can we just wrap the filtering component so that the pills and buttons are not in one line?
<img width="584" alt="Screen Shot 2022-03-09 at 16 08 05" src="https://user-images.githubusercontent.com/4550621/157472077-1fbdc370-5fd3-4dfd-8459-d4965d4e1291.png">

alexkim205 committed:
Merge remote-tracking branch 'origin' into fix/events-filter

alexkim205 commented:
Appreciate the reviews here

> Do we have a design for an alternative for the pills to be implemented in a follow-up? 
> Those grey pills look out of place now. I assume that's out of scope for now?

I'm not sure, but @clarkus probably has a better idea here.

> More immediately, can we just wrap the filtering component so that the pills and buttons are not in one line? <img alt="Screen Shot 2022-03-09 at 16 08 05" width="584" src="https://user-images.githubusercontent.com/4550621/157472077-1fbdc370-5fd3-4dfd-8459-d4965d4e1291.png">

I think wrapping makes sense, but only when the pills and buttons overflow the width. I'd prefer to keep the filter header as slim as possible so that we don't offset the actual table further down.

alexkim205 commented on frontend/src/lib/components/LemonRow/LemonRow.scss:
```diff
@@ -104,7 +104,6 @@
 }
 
 .LemonRow__icon {
-    width: 1em;
```
That makes sense now why that width was defined there. I still need to support multiple icons in-line which requires the width to fit the content of the icons, so I'll extend `LemonRow__icon` with a `LemonRow__icon--multiple` to support this use case.

alexkim205 committed:
address feedback

alexkim205 committed:
only truncate text in button

alexkim205 commented:
Updated experience


https://user-images.githubusercontent.com/13460330/157539920-714d986d-b9dd-468e-852c-bc51a4ca7d39.mov

alexkim205 commented:
Frontend CI keeps failing for an unrelated reason.

clarkus reviewed (dismissed):
Looks good to me other than the pill wrapping mentioned by others. Generally we want to use a left-to-right `inline` / `inline-block` layout for these items where the filter button an adjacent input are the first items in the row. As pills / completed filters are added, they append to the right. If no space is available, they wrap to the next available line at the same offset as the first items (the inputs I mentioned). It's a sentence-like layout. 

I will put together a longer term design for the pills. We also have the related task of clarifying property event origins, so there's definitely a developing problem to address. 

Nice job, @alexkim205 🎉

alexkim205 committed:
Merge remote-tracking branch 'origin' into fix/events-filter

alexkim205 committed:
inline pills

alexkim205 commented:
Thanks everyone. @clarkus how does this look?
<img width="1059" alt="Screen Shot 2022-03-10 at 9 48 11 AM" src="https://user-images.githubusercontent.com/13460330/157724500-e0a455b3-aa09-4b13-b416-d72e68f9242d.png">

clarkus reviewed (approved):
Latest changes look great, @alexkim205

Twixes reviewed (approved):
👌
